### PR TITLE
modification to make it work under Windows 10

### PR DESCRIPTION
**Pull request by scwall**
Hi
here I looked and indeed pyctocrome works with the imports of the libraries I had to put back some patches so that everything works again on Windows 10. Tutor is now functional under windows 10 
👍 😄

### CODE REVIEW

**Comment by regisb** [comment]
Hi @scwall! I jsut realised that I had forgotten to push my changes to the Github repo. Can you please check the current master branch and see if it works on Windows as it is?

**Comment by scwall** [comment]
Here I made the tests with the modifications the only thing that doesn't work is the user's part I'm going to find out how to recover with the windows api if powershell or cmd is launched as administrator or not :)
```
File "***tutor\tutor\commands\cli.py", line 55, in cli
    if utils.get_user_id() == 0:
  File "***tutor\tutor\utils.py", line 134, in get_user_id
    return int(check_output("id", "-u").decode())
  File "***tutor\tutor\utils.py", line 201, in check_output
    return subprocess.check_output(command)
  File "***Programs\Python\Python36\lib\subprocess.py", line 356, in check_output
    **kwargs).stdout
  File "***Programs\Python\Python36\lib\subprocess.py", line 423, in run
    with Popen(*popenargs, **kwargs) as process:
  File "***Programs\Python\Python36\lib\subprocess.py", line 729, in __init__
    restore_signals, start_new_session)
  File "***Programs\Python\Python36\lib\subprocess.py", line 1017, in _execute_child
    startupinfo)
FileNotFoundError: [WinError 2] The specified file cannot be found
```
I cancel the pulls requests and do another one

**Comment by regisb** [comment]
> `FileNotFoundError: [WinError 2] The specified file cannot be found`

This is exactly the reason why I asked you earlier whether the `id -u` command was available on Windows :)  https://github.com/overhangio/tutor/issues/381#issuecomment-723448155

**Comment by scwall** [comment]
ah okay excuse me

**Comment by scwall** [comment]
Here I found the command to see if the user has launched powershell or cmd as administrator, I also made a fix for the line feed and forced the line feed /n otherwise it was causing problems during the build of images
@regisb

File: tutor/env.py [hunk 1]
```diff
@@ -231,7 +231,7 @@ def write_to(content, path):
     if isinstance(content, bytes):
         open_mode += "b"
     utils.ensure_file_directory_exists(path)
-    with open(path, open_mode, encoding="utf8") as of:
+    with open(path, open_mode, encoding="utf8",newline='\n') as of:
```
Comment (regisb) [inline tutor/env.py:234]: Can you please explain why this is necessary?

**Comment by scwall** [comment]
I added this function because windows forces the return to the line version windows (CRLF  \r\n ) instead of LF which often causes problems with linux. So I force the writing of files in LF format so that the docker can read the files properly. Example of error this causes :)
```
 => [28/33] RUN chmod a+x /openedx/bin/*                                                                                                                                                             0.4s
 => ERROR [29/33] RUN openedx-assets xmodule     && openedx-assets npm     && openedx-assets webpack --env=prod     && openedx-assets common                                                         0.3s
------
 > [29/33] RUN openedx-assets xmodule     && openedx-assets npm     && openedx-assets webpack --env=prod     && openedx-assets common:
#33 0.337 /usr/bin/env: 'python\r': No such file or directory
------
failed to solve with frontend dockerfile.v0: failed to build LLB: executor failed running [/bin/sh -c openedx-assets xmodule     && openedx-assets npm     && openedx-assets webpack --env=prod     && ope
nedx-assets common]: runc did not terminate sucessfully
```
After forcing the carriage return the files are written in unix format :) 

image
**after** 
image

**Comment by regisb** [comment]
@scwall Thanks for your answer. Can you please try the Tutor master branch, where I added your suggested change?

**Comment by scwall** [comment]
here I added the changes and I replied to the message on openedx :D . 
Is there a discord or a chat or I could contact you ? it would be easier.

**Comment by regisb** [comment]
I prefer to keep code conversation public :) Also, what's related to this PR should remain here.

**Comment by regisb** [comment]
Can we close this? I believe all the changes are now available in the master branch.

**Comment by scwall** [comment]
Yes it is perfect if I detect other errors I will come to indicate them in a new post.

**Pull request closed without merging**